### PR TITLE
[ACA-1679] Add remaining tests for page titles

### PR DESCRIPTION
**Pull request by DenisPrivistirescu**
## PR Checklist
Please check if your PR fulfills the following requirements:

```
- [ ] The commit message follows our guidelines: https://github.com/Alfresco/alfresco-content-app/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
```

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application / Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by pionnegru** [comment]
@DenisPrivistirescu can we use async/await?

File: e2e/components/header/header.ts [hunk 1]
```diff
@@ -26,17 +26,28 @@
...
     logoLink: ElementFinder = this.component.element(this.locators.logoLink);
     userInfo: UserInfo = new UserInfo(this.component);
+    searchIcon: ElementFinder = this.component.element(this.locators.searchIcon);
+    searchBar: ElementFinder = this.component.element(this.locators.searchBar);
 
     constructor(ancestor?: ElementFinder) {
         super('adf-layout-header', ancestor);
     }
+
+    enterText (name: string) {
```
Comment (adinapitul) [inline e2e/components/header/header.ts]: method and parameter names can be improved

File: e2e/components/header/header.ts [hunk 2]
```diff
@@ -26,17 +26,28 @@
...
+        return this.searchBar.clear()
+            .then(() => Utils.typeInField(this.searchBar, name));
```
Comment (adinapitul) [inline e2e/components/header/header.ts]: why not use sendKeys?

File: e2e/suites/application/page-titles.test.ts [hunk 3]
```diff
@@ -25,13 +25,23 @@
 
 import { browser } from 'protractor';
 
-import { SIDEBAR_LABELS } from '../../configs';
+import { SIDEBAR_LABELS, PAGE_TITLES } from '../../configs';
 import { LoginPage, LogoutPage, BrowsingPage } from '../../pages/pages';
+import { RepoClient } from '../../utilities/repo-client/repo-client';
+import { Utils } from '../../utilities/utils';
+import { protractor } from '../../../node_modules/protractor/built/ptor';
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: please remove this

File: e2e/suites/application/page-titles.test.ts [hunk 4]
```diff
@@ -25,13 +25,23 @@
...
 
 describe('Page titles', () => {
     const loginPage = new LoginPage();
     const logoutPage = new LogoutPage();
     const page = new BrowsingPage();
+    const apis = {
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: you only have a single api here
const adminApi = new RepoClient() would be enough

File: e2e/suites/application/page-titles.test.ts [hunk 5]
```diff
@@ -25,13 +25,23 @@
...
+        admin: new RepoClient()
+    };
+    const { nodes: nodesApi } = apis.admin;
+    const file = `file-${Utils.random()}.txt`; let fileId;
+    const header = page.header;
+    const enterText = `text-${Utils.random()}`;
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: not needed. you can search for the file name

File: e2e/suites/application/page-titles.test.ts [hunk 6]
```diff
@@ -124,5 +134,53 @@ describe('Page titles', () => {
...
+        });
+
+        afterAll(done => {
+            logoutPage.load()
+                .then(done);
+        })
+
+        it('Search Results page - [C280413]', () => {
+            header.searchIcon.click()
+            .then(() => page.dataTable.waitForHeader())
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: fix indentation

File: e2e/suites/application/page-titles.test.ts [hunk 7]
```diff
@@ -124,5 +134,53 @@ describe('Page titles', () => {
...
+            .then(() => header.searchBar.sendKeys(enterText))
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: why not use the method you wrote in header.ts ?

File: e2e/components/header/header.ts [hunk 8]
```diff
@@ -26,17 +26,28 @@
...
 import { Component } from '../component';
 import { UserInfo } from './user-info';
+import { Utils } from '../../utilities/utils';
 
 export class Header extends Component {
     private locators = {
         logoLink: by.css('.app-menu__title'),
-        userInfo: by.css('aca-current-user')
+        userInfo: by.css('aca-current-user'),
+        searchIcon: by.css("button[class*='adf-search-button']"),
```
Comment (pionnegru) [inline e2e/components/header/header.ts]: why do we use `*`? partial class name is used?

File: e2e/suites/application/page-titles.test.ts [hunk 9]
```diff
@@ -124,5 +134,53 @@ describe('Page titles', () => {
                     expect(browser.getTitle()).toContain(label);
                 });
         });
+
+    });
+
+    describe('on viewer mode', () => {
```
Comment (pionnegru) [inline e2e/suites/application/page-titles.test.ts]: this sounds like an action not a description

File: e2e/components/header/header.ts [hunk 10]
```diff
@@ -26,17 +26,29 @@
 import { ElementFinder, by } from 'protractor';
 import { Component } from '../component';
 import { UserInfo } from './user-info';
+import { protractor } from '../../../node_modules/protractor/built/ptor';
```
Comment (adinapitul) [inline e2e/components/header/header.ts]: this is not the correct import for { protractor }

File: e2e/components/header/header.ts [hunk 11]
```diff
@@ -26,17 +26,29 @@
...
 
 export class Header extends Component {
     private locators = {
         logoLink: by.css('.app-menu__title'),
-        userInfo: by.css('aca-current-user')
+        userInfo: by.css('aca-current-user'),
+        searchIcon: by.css("button[class*='adf-search-button']"),
```
Comment (adinapitul) [inline e2e/components/header/header.ts]: it's a button, not just an icon, please name it accordingly

File: e2e/components/header/header.ts [hunk 12]
```diff
@@ -26,17 +26,29 @@
...
     logoLink: ElementFinder = this.component.element(this.locators.logoLink);
     userInfo: UserInfo = new UserInfo(this.component);
+    searchIcon: ElementFinder = this.component.element(this.locators.searchIcon);
+    searchBar: ElementFinder = this.component.element(this.locators.searchBar);
 
     constructor(ancestor?: ElementFinder) {
         super('adf-layout-header', ancestor);
     }
+
+    searchInput (textinput: string) {
```
Comment (adinapitul) [inline e2e/components/header/header.ts]: the name of the method needs improving
also, please use camel case for the parameter name

File: e2e/suites/application/page-titles.test.ts [hunk 13]
```diff
@@ -25,13 +25,21 @@
...
 import { LoginPage, LogoutPage, BrowsingPage } from '../../pages/pages';
+import { RepoClient } from '../../utilities/repo-client/repo-client';
+import { Utils } from '../../utilities/utils';
+
 
 describe('Page titles', () => {
     const loginPage = new LoginPage();
     const logoutPage = new LogoutPage();
     const page = new BrowsingPage();
+    const adminapi = new RepoClient()
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: please use camel case for variables names

File: e2e/suites/application/page-titles.test.ts [hunk 14]
```diff
@@ -124,5 +132,47 @@ describe('Page titles', () => {
...
                 });
         });
+
     });
-});
+
+    describe('on viewer mode', () => {
+        beforeAll( async (done) => {
+           const resp = (await nodesApi.createFile(file));
+            fileId = resp.data.entry.id
```
Comment (adinapitul) [inline e2e/suites/application/page-titles.test.ts]: instead of these 2 lines you can just write
`fileId = (await nodesApi.createFile(file)).entry.id;`

**Comment by adinapitul** [comment]
Please see the errors from Travis

**Pull request merged**